### PR TITLE
fix(config): don't initialize GCS state sync for localnet configs

### PR DESCRIPTION
**Pull request by marcelo-gonzalez**
The config generated by `neard init` gives the same `.state_sync` field for invocations without the `--chain-id` option set as invocations with `--chain-id mainnet`. This default value indicates that the node should try to download state parts from GCS. But this of course makes no sense for localnet tests, where all state is local to the machine running the test. Before this change, nodes in localnet tests that need to state sync (because for example a chunk producer has been kicked and the assignments across shards has changed) actually attempt to download state parts from GCS, and of course fail to do so.

So fix it by only setting this default value when `--chain-id mainnet` or `--chain-id testnet` is given, and not setting it otherwise. Then we add an `extra_state_dumper` option to the pytest `start_cluster()` and `init_cluster()` functions that will add an extra node to the test that will dump state to a local directory, and set the state sync configs in the other nodes to point to that directory

### CODE REVIEW

**Comment by marcelo-gonzalez** [comment]
for now i made this `extra_state_dumper` option default to false in case it would break some test. And I changed one test, transactions.py, to use it. I have seen that one stall many times in the past because of state sync failures, so this should actually make it more reliable. We can change more tests to use this, and then eventually maybe make it default True. And then hopefully remove the whole thing when we dont use centralized state sync anymore

**Comment by tayfunelmas** [comment]
is the state sync configured separately for "mocknet" (I guess it is handled in `mocknet.py`)?

File: pytest/lib/cluster.py [hunk 1]
```diff
@@ -900,6 +905,20 @@ def init_cluster(
 
     logger.info("Search for stdout and stderr in %s" % node_dirs)
 
+    # if extra_state_dumper is True, we added 1 to num_observers above and we will enable
+    # state dumping to a local tmp dir on the last node in node_dirs. The other nodes will have their
+    # state_sync configs point to this tmp dir
+    # TODO: remove this extra_state_dumper option when centralized state sync is no longer used
+    if extra_state_dumper:
+        (node_config_dump,
+         node_config_sync) = state_sync_lib.get_state_sync_configs_pair(
```
Comment (tayfunelmas) [inline pytest/lib/cluster.py:914]: as follow up, I think we can start using `extra_state_dumper` for all the tests that use `get_state_sync_config_pairs` now.

**Review by tayfunelmas — APPROVED** [review]
LGTM.
added a comment just to make sure "mocknet" is not affected by setting config to none for non-prod chains.

**Pull request merged**